how-it-works/1_protocol/03_message-routing.card.md

```diff
@@ -6,6 +6,12 @@ title: Message routing
 
 # Message routing
 
-The Internet Computer blockchain enables users to send messages to canister smart contracts and canisters to send messages between themselves. For scalability, the Internet Computer is composed of many subnet blockchains and the Internet Computer's Network Nervous System can add new subnets as required. The message routing component routes messages to and from canisters across all of the Internet Computer's subnet blockchains and ensures that new subnets can be added seamlessly.
+The message routing layer of the core IC protocol implements, as the name suggests, the routing of inter-canister messages between subnets, so-called cross-subnet messaging (XNet messaging).
+*Secure XNet messaging* is a key ingredient for the architecture of loosely-coupled subnets and thus a prerequisite for the scalability of the IC.
+In every IC round, message routing receives a block of canister messages to be executed from consensus - the same on each node of the subnet - and places the messages into the input queues of their target canisters, a process called *induction*. Then, it triggers the execution round which will potentially lead to new canister messages in the executed canisters' output queues. Once execution is done, these messages in the output queues will be routed to the respective subnet-to-subnet streams to be picked up by block makers on other subnets.
+Another crucial functionality is *state certification*, that is, the subnet certifying parts of the replicated subnet state in every round in a completely decentralized manner. Among others, this certification is used by other subnets to verify the authenticity of the subnet-to-subnet streams or to authentically export the status of Ingress messages.
```

Comment:
This state certification business maybe belongs in "go deeper". Also, it kind of replicates some of what is discussed in the chain-key crypto section. Not sure what to do about that. In any case, it is kinda weird that we talk about state certification here without mentioning chain-key crypto, which is an essential ingredient.

Bear in mind: a reader who is at this point in the text has no concept of chain-key crypto / threshold crypto. 

Maybe mention chain-key crypto here somehow...even better, provide a hyperlink.